crates/dbsp/src/trace/ord/fallback/indexed_wset.rs

```diff
@@ -381,80 +449,127 @@ where
         B: BatchReader,
         I: IntoIterator<Item = &'a B> + Clone,
     {
-        let cap = batches.clone().into_iter().map(|b| b.len()).sum();
         Self {
             factories: factories.clone(),
-            inner: match pick_merge_destination(batches, location) {
-                BatchLocation::Memory => BuilderInner::Vec(VecIndexedWSetBuilder::with_capacity(
-                    &factories.vec_indexed_wset_factory,
-                    cap,
-                )),
-                BatchLocation::Storage => {
-                    BuilderInner::File(FileIndexedWSetBuilder::with_capacity(factories, cap))
-                }
-            },
+            inner: BuilderInner::new(
+                factories,
+                batches.clone().into_iter().map(|b| b.len()).sum(),
+                pick_merge_destination(batches, location).into(),
+            ),
         }
     }
 
     fn push_time_diff(&mut self, time: &(), weight: &R) {
         match &mut self.inner {
             BuilderInner::Vec(vec) => vec.push_time_diff(time, weight),
             BuilderInner::File(file) => file.push_time_diff(time, weight),
+            BuilderInner::Threshold { vec, remaining } => {
+                vec.push_time_diff(time, weight);
+                if Self::should_spill(weight.size_of().total_bytes(), remaining) {
```

Comment:
This can be expensive to check on each `push_time_diff`. Might make sense to unconditionally reduce `remaining` here.

Reply:
I pushed a change for that.

Reply:
Done.